taker/src/main.rs

```diff
@@ -298,6 +319,17 @@ async fn main() -> Result<()> {
 
     let (projection_actor, projection_context) = xtra::Context::new(None);
 
+    let possible_addresses = resolve_maker_addresses(maker_url.as_str()).await?;
+
+    // Assume that the first resolved ipv4 address is good enough for libp2p.
+    let first_maker_address = possible_addresses
+        .iter()
+        .find(|x| x.is_ipv4())
+        .context("Could not resolve maker URL")?;
```

Comment:
Instead of doing this, we should wrap the `TokioTcpConfig` in a DNS transport and let it do the name resolution. See https://docs.rs/libp2p-dns/0.32.1/libp2p_dns/.

Reply:
I'll create an issue from this and deal with it in a follow-up, good idea!

Reply:
tracked by #1896